test/libsolidity/syntaxTests/nameAndTypeResolution/free_and_constant.sol

```diff
@@ -0,0 +1,4 @@
+uint constant c = 7;
+function c() returns (uint) {}
+// ----
+// DeclarationError 2333: (21-51): Identifier already declared.
```

Comment:
I hope this means that *all* conflicts of top-level construct are covered now. Probably, the function case was just special due to overload resolution and stuff, so it should be fine... theoretically we could have 42 distinct test cases like this to really cover everything explicitly, but I guess it's fine (and granted a handful of those cases probably already have other tests) :-).